beamer/events.py

```diff
@@ -214,8 +213,13 @@ def fetch(self) -> list[Event]:
                 result.extend(events)
                 from_block = BlockNumber(to_block + 1)
 
-        self._next_block_number = from_block
+        try:
+            # Block number needs to be decremented here, because it is already incremented above
+            block_data = self._contract.web3.eth.get_block(from_block - 1)
+        except requests.exceptions.RequestException:
+            return []
 
+        self._next_block_number = from_block
         result.append(
             LatestBlockUpdatedEvent(
                 chain_id=self._chain_id,
```

Comment:
I would rather do something like this:

```
        self._next_block_number = from_block
        try:
            block_data = self._contract.web3.eth.get_block(block_number)
        except requests.exceptions.RequestException:
            return result
        else:
            result.append(LatestBlockUpdatedEvent(...))
        return result
```

So that we return all the collected events even if the last RPC call to get block data fails.

Reply:
👍